components/ai_chat/resources/untrusted_conversation_frame/components/conversation_entries/conversation_entries_utils.ts

```diff
@@ -3,7 +3,30 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this file,
 // You can obtain one at https://mozilla.org/MPL/2.0/.
 
-export const getReasoningText = (text: string) => {
+import * as Mojom from '../../../common/mojom'
+
+/**
+ * Combines events of entries if there are multiple consecutive assistant entries.
+ * @param allEntries Flat list of conversation entries
+ */
+export function groupConversationEntries(allEntries: Mojom.ConversationTurn[]): Mojom.ConversationTurn[][] {
+  // TODO: test
+  const groupedEntries: Mojom.ConversationTurn[][] = []
```

Comment:
probably too late for this but could we just add the `ConversationEvents` to the same turn?

Reply:
Maybe we could do that for the frontend. Reading through this function and adding tests, I realize I didn't quite finish deciding how to handle copy and editing and I think I need to re-do this component and re-think how the grouping is handled.